quarkus/defaults/src/main/resources/application.properties

```diff
@@ -74,6 +75,20 @@ quarkus.log.category."io.smallrye.config".level=INFO
 quarkus.management.port=8182
 quarkus.management.test-port=0
 
+# OIDC settings. These settings are required only when using external authentication providers.
+# See https://quarkus.io/guides/security-oidc-configuration-properties-reference
+quarkus.oidc.application-type=service
+quarkus.oidc.resolve-tenants-with-issuer=true
+# Default tenant (disabled by default, set this to true if you use external authentication)
+quarkus.oidc.tenant-enabled=false
```

Comment:
We could enable this conditionally with a `ConfigSourceInterceptor` that would check if any realm is using external or mixed auth, in which case it makes sense to enable the default OIDC tenant, especially if not other tenant is configured. But I'm leaving this as a follow-up improvement.